src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1015,39 +1037,48 @@ namespace cryptonote
     std::vector<tx_verification_batch_info> tx_info;
     tx_info.reserve(tx_blobs.size());
     for (size_t i = 0; i < tx_blobs.size(); i++) {
-      if (!results[i].res || already_have[i])
+      if (!results[i].valid || already_have[i])
         continue;
-      tx_info.push_back({&results[i].tx, results[i].hash, tvc[i], results[i].res});
+      tx_info.push_back({&results[i].tx, results[i].tx.hash, tvc[i], results[i].valid});
     }
     if (!tx_info.empty())
       handle_incoming_tx_accumulated_batch(tx_info, tx_relay == relay_method::block);
 
+    bool valid_events = false;
     bool ok = true;
     it = tx_blobs.begin();
     for (size_t i = 0; i < tx_blobs.size(); i++, ++it) {
-      if (!results[i].res)
+      if (!results[i].valid)
       {
         ok = false;
         continue;
       }
+      results[i].valid = false;
       if (tx_relay == relay_method::block)
         get_blockchain_storage().on_new_tx_from_block(results[i].tx);
       if (already_have[i])
         continue;
 
       const uint64_t weight = results[i].tx.pruned ? get_pruned_transaction_weight(results[i].tx) : get_transaction_weight(results[i].tx, it->blob.size());
-      ok &= add_new_tx(results[i].tx, results[i].hash, tx_blobs[i].blob, weight, tvc[i], tx_relay, relayed);
+      ok &= add_new_tx(results[i].tx, results[i].tx.hash, tx_blobs[i].blob, weight, tvc[i], tx_relay, relayed);
 
       if(tvc[i].m_verifivation_failed)
-      {MERROR_VER("Transaction verification failed: " << results[i].hash);}
+      {MERROR_VER("Transaction verification failed: " << results[i].tx.hash);}
       else if(tvc[i].m_verifivation_impossible)
-      {MERROR_VER("Transaction verification impossible: " << results[i].hash);}
+      {MERROR_VER("Transaction verification impossible: " << results[i].tx.hash);}
```

Comment:
While these changes seem OK in practice, tx.hash is not really meant to be public. It's a cache that's only valid is hash_valid is set. Encouraging this use instead of get_transaction_hash will likely bite us in the rear later. So this adds a couple dozen mostly pointless lines and makes the code more brittle IMHO, even though correct.

Reply:
Then why is the field public? `get_transaction_hash` should be a public member function that checks the private internal variable and then forwards to the global function. Otherwise, this is clearly indicating accessibility to all. 

This would seem to only "bite us" if one of these functions modifies the hash after its already been calculated once. Seems like a weird update, but yeah sure I'll just duplicate it in memory like before.